File: app/src/main/java/io/eodc/planit/fragment/OnboardingAddClassesFragment.java
package io.eodc.planit.fragment;

import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import io.eodc.planit.R;
import io.eodc.planit.adapter.ClassesAdapter;
import io.eodc.planit.model.ClassListViewModel;

/**
 * The last slide of the onboarding carousel, where the user initially adds their classes
 *
 * @author 2n
 */
public class OnboardingAddClassesFragment extends OnboardingFragment {

    @BindView(R.id.recycle_class)      RecyclerView    mRvClasses;
    @BindView(R.id.image_no_class)      ImageView       mImageNoClass;
    @BindView(R.id.text_no_class)   TextView        mTextNoClass;

    @OnClick(R.id.btn_add_class)
    void addClass() {
        if (getFragmentManager() != null) {
            ModifyClassFragment.newInstance(null)
                    .show(getFragmentManager(), null);
        }
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_onboarding_classes, container, false);
        ButterKnife.bind(this, v);
        return v;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        mRvClasses.setLayoutManager(new LinearLayoutManager(getContext()));
        ViewModelProviders.of(this).get(ClassListViewModel.class)
                .getClasses().observe(this, classes -> {
                    if (classes != null) {
                        if (mRvClasses.getAdapter() != null) {
                            ((ClassesAdapter) mRvClasses.getAdapter()).swapClassesList(classes);
                        } else {
                            mRvClasses.setAdapter(new ClassesAdapter(classes, getContext()));
                        }
                        updateNoClassIndicators(classes.size());
                    }
        });
    }

    private void updateNoClassIndicators(int count) {
        if (count > 0) {
            mTextNoClass.setVisibility(View.GONE);
            mImageNoClass.setVisibility(View.GONE);
        } else {
            mTextNoClass.setVisibility(View.VISIBLE);
            mImageNoClass.setVisibility(View.VISIBLE);
        }
    }
}

File: app/src/main/java/io/eodc/planit/fragment/OnboardingAddClassesFragment.kt
package io.eodc.planit.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import io.eodc.planit.R
import io.eodc.planit.adapter.SubjectAdapter
import io.eodc.planit.model.SubjectListViewModel
import kotlinx.android.synthetic.main.fragment_onboarding_classes.*

/**
 * The last slide of the onboarding carousel, where the user initially adds their subjects
 *
 * @author 2n
 */
class OnboardingAddClassesFragment : OnboardingFragment() {

    private fun addClass() {
        if (fragmentManager != null) {
            ModifyClassFragment.newInstance(null)
                    .show(fragmentManager!!, null)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_onboarding_classes, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        recycleSubject.layoutManager = androidx.recyclerview.widget.LinearLayoutManager(context)
        ViewModelProviders.of(this).get<SubjectListViewModel>(SubjectListViewModel::class.java)
                .subjectsObservable.observe(this,
                Observer {
                    if (it != null) {
                        if (recycleSubject.adapter != null) {
                            (recycleSubject.adapter as SubjectAdapter).swapClassesList(it)
                        } else {
                            recycleSubject.adapter = SubjectAdapter(it, context!!)
                        }
                        updateNoClassIndicators(it.size)
                    }
                })
        btnAddClass.setOnClickListener { addClass() }
    }

    private fun updateNoClassIndicators(count: Int) {
        if (count > 0) {
            textOnboardingNoClass.visibility = View.GONE
            imageNoClass.visibility = View.GONE
        } else {
            textOnboardingNoClass.visibility = View.VISIBLE
            imageNoClass.visibility = View.VISIBLE
        }
    }
}
